plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiFileStatus.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hudi;
+
+import io.trino.filesystem.FileEntry;
+import org.apache.hadoop.fs.Path;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public record HudiFileStatus(Path path, boolean isDirectory, long length, long modificationTime, long blockSize)
+{
+    public static HudiFileStatus toHudiFileStatus(FileEntry entry, long blockSize)
```

Comment:
Do we need this method? The only caller of it creates a `FileEntry` object just to call this method, then this method unpacks the object. Seems simpler to call the constructor directly with the fields.

Reply:
Removed. Instantiating `HudiFileStatus` directly.